### PR TITLE
Change to EnsureChannelFirst to include behaviour from deprecated transforms

### PR DESCRIPTION
**Pull request by ericspod**
Signed-off-by: Eric Kerfoot <eric.kerfoot@kcl.ac.uk>

### Description
`EnsureChannelFirst` should be used in place of `AddChannel`, however it requires that metadata be provided as MetaTensor inputs or separate dictionaries. This doesn't substitute for `AddChannel` which required neither of these things and so can't be used in the situation where one has data known to have no channel dimension nor metadata and we just want to hard-code adding a 1-sized channel dimension. This is useful in a lot of examples that use generated data and not loaded files that come with metadata, and avoids forcing the user to use MetaTensor.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by rijobro — APPROVED** [review]

File: monai/transforms/utility/array.py [hunk 1]
```diff
@@ -202,43 +202,57 @@ def __call__(self, img: NdarrayOrTensor) -> NdarrayOrTensor:
...
     backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
 
-    def __init__(self, strict_check: bool = True):
-        """
-        Args:
-            strict_check: whether to raise an error when the meta information is insufficient.
-        """
+    def __init__(self, strict_check: bool = True, add_channel_default=True):
         self.strict_check = strict_check
+        self.add_channel_default = add_channel_default
```
Comment (KumoLiu) [inline monai/transforms/utility/array.py:220]: I think the argument name `add_channel_default` may be a little confused, perhaps name such as  `force_add_channel` maybe more  easier to understand.

Comment (wyli) [reply monai/transforms/utility/array.py:220]: I think it provides a default value only if channel_dim is None, so it's not forcing the value...

Comment (ericspod) [reply monai/transforms/utility/array.py:220]: It doesn't force adding the channel no, but if that was desired for some reason we should undeprecate `AddChannel`.

**Pull request merged**